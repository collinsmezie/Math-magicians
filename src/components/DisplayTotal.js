import PropTypes from 'prop-types';

function DisplayTotal(props) {
  const { total, next, operation } = props;
  return (
    <div className="display">
      <h1>
        {total}
        {operation}
        {next}
      </h1>
    </div>
  );
}

DisplayTotal.propTypes = {
  total: PropTypes.string.isRequired,
  next: PropTypes.string.isRequired,
  operation: PropTypes.string.isRequired,
};

export default DisplayTotal;
