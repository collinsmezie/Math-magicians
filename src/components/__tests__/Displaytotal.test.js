import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import DisplayTotal from '../DisplayTotal';

describe('Display page of the App', () => {
  test('Builds the snapshot of the Display component safely', () => {
    const rend = renderer.create(<DisplayTotal />);
    expect(rend.toJSON()).toMatchSnapshot();
  });
  test('Render Display component', () => {
    render(<DisplayTotal />);
  });
});
