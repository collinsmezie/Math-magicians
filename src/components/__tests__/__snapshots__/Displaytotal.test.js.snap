// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display page of the App Builds the snapshot of the Display component safely 1`] = `
<div
  className="display"
>
  <h1 />
</div>
`;
